src/builders.rs

```diff
@@ -1208,3 +1240,20 @@ impl fmt::Debug for Output {
         self.0.fmt(f)
     }
 }
+
+#[derive(Debug)]
+pub struct CustomFileLoggerInfo{
```

Comment:
If you'd be wiling to make it, a new example file like `examples/date-based-file-log.rs` which uses this logger could be good both for testing and for showing users an example of using this.

It could be a pretty straight copy of `examples/syslog.rs` except using this logger instead of `syslog`.

Just to have an example to run and easily test this functionality.